src/coreclr/vm/interpexec.cpp

```diff
@@ -1081,13 +1127,25 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                     }
                     else
                     {
-                        // At this stage in the implementation, we assume this is pointer to
-                        // interpreter code. In the future, this should probably be tagged pointer
-                        // for interpreter call or normal pointer for JIT/R2R call.
                         targetIp = (const int32_t*)targetMethod;
                     }
                     }
 CALL_TARGET_IP:
+                    // HACK: we need a fast way to check of the targetIp is an interpreter code or not.
+                    // Probably use a tagged pointer for interpreter code and a normal pointer for JIT/R2R code.
+                    EECodeInfo codeInfo((PCODE)targetIp);
+                    if (!codeInfo.IsValid())
+                    {
+                        printf("Attempted to execute native code from interpreter.\n");
+                        assert(0);
```

Comment:
This should be a FAILFAST instead. Look for uses of `EEPOLICY_HANDLE_FATAL_ERROR_WITH_MESSAGE` and/or `COR_E_FAILFAST` in the VM.